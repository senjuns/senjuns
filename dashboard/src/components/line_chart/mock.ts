import { timeToEpoch, getRandomValueBetween } from 'shared/utils';
import { MeasurementTypes } from 'shared/interfaces';

export const getData = (date: string | null, measurementType: string) => {
  if (!date) return null;
  const startDate = new Date(date);
  startDate.setHours(0);
  startDate.setMinutes(0);
  startDate.setSeconds(0);

  const endDate = new Date(startDate);
  endDate.setDate(startDate.getDate() + 1);

  const measurementEpochTimes = [];
  const measurementValues = [];

  let minVal = 0;
  let maxVal = 0;
  let redLow;
  let yellowLow;
  let green;
  let yellowHigh;
  let redHigh;

  switch (measurementType) {
    case MeasurementTypes.Light:
      minVal = 50;
      maxVal = 100;
      redLow = { min: 50, max: 60 };
      yellowLow = { min: 60, max: 70 };
      green = { min: 70, max: 80 };
      yellowHigh = { min: 80, max: 90 };
      redHigh = { min: 90, max: 100 };
      break;
    case MeasurementTypes.Temperature:
      minVal = 10;
      maxVal = 50;
      redLow = { min: 10, max: 20 };
      yellowLow = { min: 20, max: 25 };
      green = { min: 25, max: 35 };
      yellowHigh = { min: 35, max: 40 };
      redHigh = { min: 40, max: 50 };
      break;
    case MeasurementTypes.Humidity:
      minVal = 30;
      maxVal = 90;
      redLow = { min: 30, max: 40 };
      yellowLow = { min: 40, max: 60 };
      green = { min: 60, max: 70 };
      yellowHigh = { min: 60, max: 80 };
      redHigh = { min: 80, max: 90 };
      break;
    case MeasurementTypes.LeafVpd:
    case MeasurementTypes.Vpd:
      minVal = 0;
      maxVal = 40;
      redLow = { min: 0, max: 10 };
      yellowLow = { min: 10, max: 15 };
      green = { min: 15, max: 25 };
      yellowHigh = { min: 25, max: 30 };
      redHigh = { min: 30, max: 40 };
      break;
    case MeasurementTypes.Co2:
      minVal = 10;
      maxVal = 100;
      redLow = { min: 10, max: 20 };
      yellowLow = { min: 30, max: 50 };
      green = { min: 50, max: 70 };
      yellowHigh = { min: 80, max: 90 };
      redHigh = { min: 90, max: 100 };
      break;
    case MeasurementTypes.Airflow:
      minVal = 0;
      maxVal = 30;
      redLow = { min: 0, max: 5 };
      yellowLow = { min: 5, max: 15 };
      green = { min: 15, max: 20 };
      yellowHigh = { min: 20, max: 25 };
      redHigh = { min: 25, max: 30 };
      break;
    case MeasurementTypes.Ppfd:
      minVal = 50;
      maxVal = 150;
      redLow = { min: 50, max: 60 };
      yellowLow = { min: 60, max: 90 };
      green = { min: 90, max: 110 };
      yellowHigh = { min: 110, max: 140 };
      redHigh = { min: 140, max: 150 };
      break;
  }

  for (let t = startDate; t <= endDate; t.setTime(t.getTime() + 10 * 60000)) {
    measurementEpochTimes.push(timeToEpoch(t.toISOString()));
    measurementValues.push(getRandomValueBetween(minVal, maxVal));
  }

  return {
    data: {
      timestamps: measurementEpochTimes,
      values: measurementValues,
    },
    min: minVal,
    max: maxVal,
    redLow,
    yellowLow,
    green,
    yellowHigh,
    redHigh,
  };
};
