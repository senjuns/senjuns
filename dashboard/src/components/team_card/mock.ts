export const teamCardsMock = [
  {
    teamName: 'AWS CDK Team',
    teamDescription:
      'we are obsessed with AWS and building cool solutions for our customers ...',
    tags: ['AWS', 'IaC', 'DevOps'],
    members: [
      {
        image: 'https://i.pravatar.cc/300',
        firstName: 'Martin',
        role: '- Senior AWS Cloud Engineer',
      },
      {
        image: 'https://i.pravatar.cc/301',
        firstName: 'Ben-Hur',
        role: '- Junior AWS Cloud Engineer',
      },
    ],
  },
  {
    teamName: 'Data Engineering Team',
    teamDescription:
      'we are a team of ex-FAAN and current FAANg employees with an average of 11 years experience building and scaling products like Instagram an Gmail...',
    tags: ['ML', 'Python', 'Panda', 'PySpark'],
    members: [
      {
        image: 'https://i.pravatar.cc/302',
        firstName: 'Peter',
        role: '- Senior AWS Cloud Engineer',
      },
      {
        image: 'https://i.pravatar.cc/303',
        firstName: 'Naga',
        role: '- Junior AWS Cloud Engineer',
      },
    ],
  },
  {
    teamName: 'React Team',
    teamDescription: 'we are nice ...',
    tags: ['AWS', 'IaC'],
    members: [
      {
        image: 'https://i.pravatar.cc/304',
        firstName: 'Ben-Hur',
        role: '- Senior React Engineer',
      },
      {
        image: 'https://i.pravatar.cc/305',
        firstName: 'Martin',
        role: '- Junior React Engineer',
      },
    ],
  },
];
