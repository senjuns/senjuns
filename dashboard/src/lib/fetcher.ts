import { GraphQLResult, GRAPHQL_AUTH_MODE } from '@aws-amplify/api/lib/types';
import { API as AmplifyAPI, graphqlOperation } from 'aws-amplify';

let instance: API | undefined = undefined;

export function amplifyFetcher<TData, TVariables>(
  query: string,
  variables?: TVariables,
) {
  return async (): Promise<TData> => {
    console.log('VARIABLES:', variables);

    const api = API.getInstance();
    const response = await api.query(query, variables);
    return response.data;
  };
}

export class API {
  protected isSignedIn: boolean = false;
  protected apiKey: string;

  constructor() {
    this.isSignedIn = false;
    this.apiKey = 'no Key';
  }

  static getInstance(): API {
    if (!instance) instance = new API();
    return instance;
  }

  static updateApiKey(apiKey: string): void {
    if (!instance) instance = new API();
    instance.apiKey = apiKey;
  }

  public getApiKey(): string {
    return this.apiKey;
  }

  static updateIsSignedIn(signedIn: boolean): void {
    if (!instance) instance = new API();
    instance.isSignedIn = signedIn;
  }

  public async query(query: string, variables?: any) {
    const operation = {
      authMode: this.isSignedIn
        ? GRAPHQL_AUTH_MODE.AMAZON_COGNITO_USER_POOLS
        : GRAPHQL_AUTH_MODE.AWS_IAM,
      ...graphqlOperation(query, variables),
    };

    // console.log(`operation=${JSON.stringify(operation)}`);

    return (await AmplifyAPI.graphql(operation)) as GraphQLResult<any>;
  }

  public async queryWithApiKey(query: string, variables?: any) {
    const operation = {
      authMode: GRAPHQL_AUTH_MODE.API_KEY,
      ...graphqlOperation(query, variables),
    };
    return (await AmplifyAPI.graphql(operation)) as GraphQLResult<any>;
  }
}
