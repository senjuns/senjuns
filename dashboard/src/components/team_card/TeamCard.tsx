// import { FC } from 'react';
import styled from 'styled-components';
import { InterBoldMirage16px, Poppins22, Poppins26 } from '../../shared/fonts';
// import { useGetLatestPhotoFeedDataBySystemId } from './useListTeamCardData';
import { useScreenSize } from '../../hooks/useScreenSize';
import { ResponsiveLayoutProps } from '../../shared/types';
import { useListTeamCardsQuery } from '../../lib/api';

const TeamCard = () => {
  const { isMobile } = useScreenSize();
  // const s = useGetLatestPhotoFeedDataBySystemId({});

  const { data, isLoading } = useListTeamCardsQuery(undefined, {
    refetchOnWindowFocus: false,
  });

  if (isLoading) return <div>Loading...</div>;

  return (
    <div>
      {data?.listTeamCards?.items?.map((teamCard, index) => (
        <div key={index} className="container-center-horizontal">
          <div className="teamcard screen">
            <Details>
              <DetailsDescription>
                <DetailsDescriptionTitle>
                  {teamCard?.teamName}
                </DetailsDescriptionTitle>
                <Tags>
                  {teamCard?.tags?.map((tag, index) => (
                    <Tag key={index}>
                      <Text>{tag}</Text>
                    </Tag>
                  ))}
                </Tags>
                <DetailsDescriptionBody>
                  {teamCard?.teamDescription}
                </DetailsDescriptionBody>
              </DetailsDescription>
            </Details>
            <Members isMobile={isMobile}>
              {teamCard?.members?.map((member, index) => (
                <Member key={index}>
                  <Image src={member?.image} />
                  <MemberDescription>
                    <FirstName>{member?.firstName}</FirstName>
                    <Role>{member?.role}</Role>
                  </MemberDescription>
                </Member>
              ))}
            </Members>
          </div>
          <hr></hr>
        </div>
      ))}
    </div>
  );
};

export default TeamCard;

const Details = styled.div`
  display: flex;
  width: 100%;
  padding: 20px;
  margin-left: 40px;
`;

const DetailsDescription = styled.div`
  display: flex;
  flex-direction: column;
`;

const DetailsDescriptionTitle = styled.h1`
  ${Poppins26}
  min-height: 30px;
  font-weight: 400;
  color: var(--black);
  line-height: 30px;
  white-space: nowrap;
`;

const DetailsDescriptionBody = styled.div`
  ${Poppins22}
  margin-top: 40px;
  font-weight: 400;
  color: var(--black);
  line-height: 30px;
`;

const Tags = styled.div`
  display: flex;
  align-items: flex-start;
`;

const Text = styled.div`
  ${InterBoldMirage16px}
  min-height: 20px;
  text-align: center;
  letter-spacing: 0.64px;
  line-height: 20px;
  white-space: nowrap;
`;

const Tag = styled.div`
  height: 44px;
  display: flex;
  padding: 0 16px;
  align-items: center;
  background-color: var(--gallery);
  border-radius: 99px;
  justify-content: center;
  margin-right: 38px;
`;

const Members = styled.div<ResponsiveLayoutProps>`
  display: flex;
  align-items: center;
  padding: ${({ isMobile }) => (isMobile ? '90px 20px 60px' : '')};
  gap: 30px;
  flex-wrap: wrap;
  justify-content: space-evenly;
  width: 100%;
`;

const Member = styled.div`
  width: 300px;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const Image = styled.img`
  width: 200px;
  height: 201px;
`;

const MemberDescription = styled.div`
  ${Poppins22}
  height: 1px;
  margin-top: 40px;
  display: flex;
  padding: 0 72.5px;
  align-items: flex-end;
  justify-content: center;
`;

const FirstName = styled.div`
  min-height: 30px;
  margin-bottom: -29px;
  font-weight: 400;
  color: var(--black);
  text-align: center;
  line-height: 30px;
  white-space: nowrap;
`;

const Role = styled.div`
  min-height: 30px;
  margin-left: 5px;
  margin-bottom: -29px;
  font-weight: 400;
  color: var(--black);
  text-align: center;
  line-height: 30px;
  white-space: nowrap;
`;
