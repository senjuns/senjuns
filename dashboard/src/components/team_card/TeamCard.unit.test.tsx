// eslint-disable-next-line import/no-extraneous-dependencies
import { QueryClient, QueryClientProvider } from 'react-query';
// import * as ShallowRenderer from 'react-test-renderer/shallow';
import renderer from 'react-test-renderer';
import { ListTeamCardsQuery } from '../../lib/api';
import TeamCard from './TeamCard';
import { teamCardsMock } from './mock';

const queryClient = new QueryClient();

const listTeamCards: ListTeamCardsQuery = {
  listTeamCards: {
    items: teamCardsMock.map((teamCard, index) => {
      return {
        id: index.toString(),
        teamName: teamCard.teamName,
        tags: teamCard.tags,
        teamDescription: teamCard.teamDescription,
        members: teamCard.members,
        createdAt: new Date(),
        updatedAt: new Date(),
      };
    }),
  },
};

const mockListTeamCardQuery = { data: listTeamCards, isLoading: false };

jest.mock('../../lib/api', () => ({
  useListTeamCardsQuery: () => {
    return mockListTeamCardQuery;
  },
}));

describe('<TeamCard />', () => {
  // it('renders correctly.', async () => {
  //   const utils = ShallowRenderer.createRenderer();
  //   utils.render(
  //     <QueryClientProvider client={queryClient}>
  //       <TeamCard />
  //     </QueryClientProvider>,
  //   );

  //   const view = utils.getRenderOutput();
  //   expect(view).toMatchSnapshot();
  // });

  it('deep renders correctly.', async () => {
    const tree = renderer
      .create(
        <QueryClientProvider client={queryClient}>
          <TeamCard />
        </QueryClientProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
