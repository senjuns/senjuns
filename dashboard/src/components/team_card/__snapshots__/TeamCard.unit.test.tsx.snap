// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamCard /> deep renders correctly. 1`] = `
<div>
  <div
    className="container-center-horizontal"
  >
    <div
      className="teamcard screen"
    >
      <div
        className="sc-bdvvtL gizfTv"
      >
        <div
          className="sc-gsDKAQ gFZcdC"
        >
          <h1
            className="sc-dkPtRN eYYopd"
          >
            AWS CDK Team
          </h1>
          <div
            className="sc-eCImPb hGygE"
          >
            <div
              className="sc-gKclnd egdtCv"
            >
              <div
                className="sc-jRQBWg gzcDxS"
              >
                AWS
              </div>
            </div>
            <div
              className="sc-gKclnd egdtCv"
            >
              <div
                className="sc-jRQBWg gzcDxS"
              >
                IaC
              </div>
            </div>
            <div
              className="sc-gKclnd egdtCv"
            >
              <div
                className="sc-jRQBWg gzcDxS"
              >
                DevOps
              </div>
            </div>
          </div>
          <div
            className="sc-hKwDye fzKgPb"
          >
            we are obsessed with AWS and building cool solutions for our customers ...
          </div>
        </div>
      </div>
      <div
        className="sc-iCfMLu bcfidr"
      >
        <div
          className="sc-furwcr fUWVbG"
        >
          <img
            className="sc-pVTFL fziFaN"
            src="https://i.pravatar.cc/300"
          />
          <div
            className="sc-jrQzAO fRGqez"
          >
            <div
              className="sc-kDTinF bHAUvf"
            >
              Martin
            </div>
            <div
              className="sc-iqseJM cqKyoS"
            >
              - Senior AWS Cloud Engineer
            </div>
          </div>
        </div>
        <div
          className="sc-furwcr fUWVbG"
        >
          <img
            className="sc-pVTFL fziFaN"
            src="https://i.pravatar.cc/301"
          />
          <div
            className="sc-jrQzAO fRGqez"
          >
            <div
              className="sc-kDTinF bHAUvf"
            >
              Ben-Hur
            </div>
            <div
              className="sc-iqseJM cqKyoS"
            >
              - Junior AWS Cloud Engineer
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr />
  </div>
  <div
    className="container-center-horizontal"
  >
    <div
      className="teamcard screen"
    >
      <div
        className="sc-bdvvtL gizfTv"
      >
        <div
          className="sc-gsDKAQ gFZcdC"
        >
          <h1
            className="sc-dkPtRN eYYopd"
          >
            Data Engineering Team
          </h1>
          <div
            className="sc-eCImPb hGygE"
          >
            <div
              className="sc-gKclnd egdtCv"
            >
              <div
                className="sc-jRQBWg gzcDxS"
              >
                ML
              </div>
            </div>
            <div
              className="sc-gKclnd egdtCv"
            >
              <div
                className="sc-jRQBWg gzcDxS"
              >
                Python
              </div>
            </div>
            <div
              className="sc-gKclnd egdtCv"
            >
              <div
                className="sc-jRQBWg gzcDxS"
              >
                Panda
              </div>
            </div>
            <div
              className="sc-gKclnd egdtCv"
            >
              <div
                className="sc-jRQBWg gzcDxS"
              >
                PySpark
              </div>
            </div>
          </div>
          <div
            className="sc-hKwDye fzKgPb"
          >
            we are a team of ex-FAAN and current FAANg employees with an average of 11 years experience building and scaling products like Instagram an Gmail...
          </div>
        </div>
      </div>
      <div
        className="sc-iCfMLu bcfidr"
      >
        <div
          className="sc-furwcr fUWVbG"
        >
          <img
            className="sc-pVTFL fziFaN"
            src="https://i.pravatar.cc/302"
          />
          <div
            className="sc-jrQzAO fRGqez"
          >
            <div
              className="sc-kDTinF bHAUvf"
            >
              Peter
            </div>
            <div
              className="sc-iqseJM cqKyoS"
            >
              - Senior AWS Cloud Engineer
            </div>
          </div>
        </div>
        <div
          className="sc-furwcr fUWVbG"
        >
          <img
            className="sc-pVTFL fziFaN"
            src="https://i.pravatar.cc/303"
          />
          <div
            className="sc-jrQzAO fRGqez"
          >
            <div
              className="sc-kDTinF bHAUvf"
            >
              Naga
            </div>
            <div
              className="sc-iqseJM cqKyoS"
            >
              - Junior AWS Cloud Engineer
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr />
  </div>
  <div
    className="container-center-horizontal"
  >
    <div
      className="teamcard screen"
    >
      <div
        className="sc-bdvvtL gizfTv"
      >
        <div
          className="sc-gsDKAQ gFZcdC"
        >
          <h1
            className="sc-dkPtRN eYYopd"
          >
            React Team
          </h1>
          <div
            className="sc-eCImPb hGygE"
          >
            <div
              className="sc-gKclnd egdtCv"
            >
              <div
                className="sc-jRQBWg gzcDxS"
              >
                AWS
              </div>
            </div>
            <div
              className="sc-gKclnd egdtCv"
            >
              <div
                className="sc-jRQBWg gzcDxS"
              >
                IaC
              </div>
            </div>
          </div>
          <div
            className="sc-hKwDye fzKgPb"
          >
            we are nice ...
          </div>
        </div>
      </div>
      <div
        className="sc-iCfMLu bcfidr"
      >
        <div
          className="sc-furwcr fUWVbG"
        >
          <img
            className="sc-pVTFL fziFaN"
            src="https://i.pravatar.cc/304"
          />
          <div
            className="sc-jrQzAO fRGqez"
          >
            <div
              className="sc-kDTinF bHAUvf"
            >
              Ben-Hur
            </div>
            <div
              className="sc-iqseJM cqKyoS"
            >
              - Senior React Engineer
            </div>
          </div>
        </div>
        <div
          className="sc-furwcr fUWVbG"
        >
          <img
            className="sc-pVTFL fziFaN"
            src="https://i.pravatar.cc/305"
          />
          <div
            className="sc-jrQzAO fRGqez"
          >
            <div
              className="sc-kDTinF bHAUvf"
            >
              Martin
            </div>
            <div
              className="sc-iqseJM cqKyoS"
            >
              - Junior React Engineer
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr />
  </div>
</div>
`;
